jcl/src/java.base/share/classes/java/lang/StackTraceElement.java

```diff
@@ -157,8 +180,57 @@ public String getModuleName() {
 public String getModuleVersion() {
 	return moduleVersion;
 }
+
+/**
+ * Returns whether the classloader name should be included in the stack trace for the provided StackTraceElement.
+ *
+ * @return true if the classloader name should be included, false otherwise.
+ */
+boolean getIncludeClassLoaderName() {
+	return includeClassLoaderName;
+}
+
+/**
+ * Returns whether the module version should be included in the stack trace for the provided StackTraceElement.
+ *
+ * @return true if the module version should be included, false otherwise.
+ */
+boolean getIncludeModuleVersion() {
+	return includeModuleVersion;
+}
+
+/**
+ * Set the includeClassLoaderName and includeModuleVersion fields for this StackTraceElement.
+ *
+ * @param classLoader the classloader for the StackTraceElement.
+ */
+void setIncludeInfoFlags(ClassLoader classLoader) {
+	/**
+	 * If the classloader is one of the Platform or Bootstrap built-in classloaders,
+	 * don't include its name or module version in the stack trace. If it is the
+	 * Application/System built-in classloader, don't include the class name, but
+	 * include the module version.
+	 */
+	if ((null == classLoader)
+		|| (ClassLoader.getPlatformClassLoader() == classLoader) // VM: Extension ClassLoader
+		|| (ClassLoader.bootstrapClassLoader == classLoader) // VM: System ClassLoader
+	) {
+		includeClassLoaderName = false;
+		includeModuleVersion = false;
+	} else if ((ClassLoader.getSystemClassLoader() == classLoader)) { // VM: Application ClassLoader
+		includeClassLoaderName = false;
+	}
+}
+
+/**
+ * Disable including the classloader name and the module version for this StackTraceElement.
+ */
+void disableIncludeInfoFlags() {
+	includeClassLoaderName = false;
+	includeModuleVersion = false;
+}
```

Comment:
Temporary function used by `StackWalker` to keep the existing behaviour for stack trace output.